### PR TITLE
bot: Single Approval Paths

### PR DESCRIPTION
**Pull request by jimbishopp**
This PR addresses the "Single Approver" issue described in #205.

Add the ability for certain PRs to require a single approval. 

A new setting has been added, singleApproverPaths, that defines paths in a repository that only require a single approver. If all the files in a PR only match paths defined in singleApproverPaths, then the PR only requires a single approval. Otherwise, the PR requires the default number of approvals, which is currently defined as 2.

A second setting was added, singleApproverAuthors, that defines PR authors that only require a single approval (dependency bots in cloud for now).

### CODE REVIEW

File: bot/internal/bot/bot.go
Comment (jentfoo) [inline bot/internal/bot/bot.go:167]: Currently this will match partial filenames, for example a allowed prefix of `src/dir` would also match to `src/dir2`.  Maybe not concerning, but I wanted to mention it.

**Review by jentfoo — APPROVED** [review]

File: bot/internal/bot/bot.go
Comment (jimbishopp) [reply bot/internal/bot/bot.go:167]: Thanks. This is by design and is the same as what we do for preferred reviewer paths.

**Comment by jimbishopp** [comment]
I'm thinking it may be better to remove the single approver paths from the config file and just define it as a go literal.

**Comment by jimbishopp** [comment]
I added another commit that allows certain authors such as dependabot or renovate to only require a single approver.

File: bot/internal/bot/bot.go
Comment (zmb3) [inline bot/internal/bot/bot.go:177]:
```suggestion
	if slices.Contains(authors, author) {
		return 1
	}
```

File: bot/internal/bot/bot.go
Comment (zmb3) [inline bot/internal/bot/bot.go:167]: Should we enforce some sort of naming convention for bots that only require a single approval as an extra precaution? I don't think we'd ever want an employee to unconditionally only require a single approval.

Alternatively, we can just require that any author configured as a single approval also passes `isAllowedRobot`.

File: bot/internal/bot/bot.go
Comment (zmb3) [inline bot/internal/bot/bot.go:185]: nit: `slices.ContainsFunc` might clean this loop up.

**Review by zmb3 — APPROVED** [review]

File: bot/internal/bot/bot.go
Comment (jimbishopp) [reply bot/internal/bot/bot.go]: My muscle memory hasn't gotten used to the slices package yet. I need to code more :/

File: bot/internal/bot/bot.go
Comment (jimbishopp) [reply bot/internal/bot/bot.go:168]: Good idea. I added a test to ensure only bots can be configured as single approver authors. We can also export `isAllowedRobot` from the `review` package and test in `approverCount` if you think we should have a secondary set of checks.

**Pull request merged**
